### PR TITLE
[2.0] Simplify IntegrationSettingsCollection

### PR DESCRIPTION
**Pull request by andrewlock**
This PR removes the `ConcurrentDictionary` associated with `ImmutableTracerSettingsCollection` and `TracerSettingsCollection`, deletes the now-obsolete `IntegrationInfo`, and switches to using `IntegrationIds` directly.

> This PR requires #1951 to be merged first

#1951 introduced `ImmutableTracerSettingsCollection`, and it was pointed out that the `ConcurrentDictionary<>` in use both there and in the existing `TracerSettingsCollection` is [no longer required](https://github.com/DataDog/dd-trace-dotnet/pull/1951#discussion_r749437977) thanks to the version conflict resolution. 

> This was meant to support looking up integrations by name if there was a version mismatch between Datadog.Trace.dll and Datadog.Trace.ClrProfiler.Managed.dll (i.e. if a new integration didn't exist in the older dll).
#1016 (comment)
>
>Since this type of version mismatch can no longer happen, I don't think we need the string-based API at all anymore

If a customer tries to fetch a non-existent integration, we log a warning, and return a "default" settings value:

```csharp
var tracerSettings = TracerSettings.FromDefaultSources();

// Access settings for unknown integration
// logs a warning, but does not throw
var settings = tracerSettings .Integrations["MyRandomIntegration"];

// changes are not persisted
settings.Enabled = false; 

bool? isEnabled = tracerSettings .Integrations["MyRandomIntegration"].Enabled;

// The change has not been persisted
Assert.Null(isEnabled);
```

I kept a similar indexer api for `ImmutableTracerSettingsCollection` but we _could_ remove this entirely?

@DataDog/apm-dotnet

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/ImmutableIntegrationSettingsCollection.cs [hunk 1]
```diff
@@ -15,64 +13,52 @@ namespace Datadog.Trace.Configuration
...
+                    return Settings[(int)integrationId];
+                }
+
+                Log.Warning(
+                    "Accessed integration settings for unknown integration {IntegrationName}. Returning default settings",
+                    integrationName);
+
+                return new ImmutableIntegrationSettings(integrationName);
             }
         }
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Configuration/ImmutableIntegrationSettingsCollection.cs:52]: Do we still need to keep this api? Probably best to I guess, but we shouldn't use it internally

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will will **increase** line coverage by `4%`
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will **not change** branch coverage
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will will **decrease** complexity by `563`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) | #2048       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `10494` / `15636` | `9094` / `12804` |          |
| Lines %   | `67%`      | `71%`      |  `4%` :heavy_check_mark:  |
| Branches  | `4589` / `6890` | `4070` / `6134` |          |
| Branches %| `67%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `8098`      | `7535`        |  `-563`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81337&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81337/artifacts?artifactName=Code%20Coverage%20Report_81337&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81426&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81426/artifacts?artifactName=Code%20Coverage%20Report_81426&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) | #2048       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `67%`      | `71%`       |  `4%` :heavy_check_mark:  |
| Branches %| `67%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `8098`      | `7535`       |  `-563` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.DuckTyping.DuckTypeProxyMethodsWithGenericParametersNotSupportedInNonPublicInstancesException](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.DuckTyping.DuckTypeTypeIsNotPublicException](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.PlatformHelpers.ServiceFabric](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/PlatformHelpers/ServiceFabric.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `-1` :heavy_check_mark: |
| [Datadog.Trace.AppSec.Transports.Http.IpInfo](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/AppSec/Transports/Http/IpInfo.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.ClrProfiler.Helpers.PointerHelpers](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/ClrProfiler/Helpers/PointerHelpers.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.AppSec.InstrumentationGatewayEventArgs](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/AppSec/InstrumentationGatewayEventArgs.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpRequest](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/HttpOverStreams/HttpRequest.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpResponse](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/HttpOverStreams/HttpResponse.cs) | `-50%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.Transports.HttpStreamResponse](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/Agent/Transports/HttpStreamResponse.cs) | `-33%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.SpanContext](https://github.com/DataDog/dd-trace-dotnet/tree/cc621eb7c78afad4825687b9f06e0f7c9c922fe8/tracer/src/Datadog.Trace/SpanContext.cs) | `-27%` :no_entry: | `0%` :heavy_check_mark:   | `2` :no_entry: |
| ...And 50 more  | | | |


The following classes were added in #2048:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.AppSec.AspNetCore.BlockingMiddleware](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `41%` | `19%` | `25` |
| [Datadog.Trace.AppSec.BlockActionException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.AppSec.PageBlockedByAppSecException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.Ci.CITracerManager](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `1` |
| [Datadog.Trace.Ci.CITracerManagerFactory](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `3` |
| ...And 36 more  | | | |


76 classes were removed from Datadog.Trace in #2048


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81337&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81337/artifacts?artifactName=Code%20Coverage%20Report_81337&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81426&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81426/artifacts?artifactName=Code%20Coverage%20Report_81426&api-version=6.0&%24format=zip)

File: tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace Datadog.Trace.AspNet
     [EditorBrowsable(EditorBrowsableState.Never)]
     public class TracingHttpModule : IHttpModule
     {
-        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(nameof(IntegrationIds.AspNet));
+        internal static readonly IntegrationIds IntegrationId = IntegrationIds.AspNet;
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs:32]: Can we rename the enum to `IntegrationId` (singular)?

> ✔️ DO use a singular type name for an enumeration unless its values are bit fields.

from https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations

File: tracer/src/Datadog.Trace/TracerManager.cs [hunk 3]
```diff
@@ -292,12 +292,15 @@ private static async Task WriteDiagnosticLog(TracerManager instance)
                     writer.WritePropertyName("disabled_integrations");
                     writer.WriteStartArray();
 
-                    // For consistency with previous diagnostic log, this only includes integrations
-                    // that were disabled with DD_DISABLED_INTEGRATIONS, not integrations disabled with
-                    // DD_INTEGRATION_{0}_DISABLED
-                    foreach (var integration in instanceSettings.Integrations.DisabledIntegrations)
+                    // In contrast to 1.x, this only shows _known_ integrations, but
+                    // lists them whether they were disabled with DD_DISABLED_INTEGRATIONS,
+                    // or DD_INTEGRATION_{0}_DISABLED
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/TracerManager.cs:297]:
```suggestion
                    // or DD_TRACE_{0}_DISABLED
```

**Review by lucaspimentel — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

File: tracer/src/Datadog.Trace/TracerManager.cs (hunk 3 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace/TracerManager.cs]: Updated in https://github.com/DataDog/dd-trace-dotnet/pull/2048/commits/b464c316b326895ddb908f14421ca58a14d71a86

File: tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs (hunk 2 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs]: This had been bugging me too 😄 Changed in https://github.com/DataDog/dd-trace-dotnet/pull/2048/commits/ddbcbea66e6f38eb7dd461a9eefaf3fc0d989824

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will will **increase** line coverage by `4%`
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will **not change** branch coverage
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) will will **decrease** complexity by `563`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) | #2048       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `10494` / `15636` | `9092` / `12804` |          |
| Lines %   | `67%`      | `71%`      |  `4%` :heavy_check_mark:  |
| Branches  | `4589` / `6890` | `4068` / `6134` |          |
| Branches %| `67%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `8098`      | `7535`        |  `-563`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81337&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81337/artifacts?artifactName=Code%20Coverage%20Report_81337&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81545&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81545/artifacts?artifactName=Code%20Coverage%20Report_81545&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f2036cf8797b1a0886555f9f4cbb8aab22c9a93) | #2048       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `67%`      | `71%`       |  `4%` :heavy_check_mark:  |
| Branches %| `67%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `8098`      | `7535`       |  `-563` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.DuckTyping.DuckTypeProxyMethodsWithGenericParametersNotSupportedInNonPublicInstancesException](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.DuckTyping.DuckTypeTypeIsNotPublicException](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.PlatformHelpers.ServiceFabric](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/PlatformHelpers/ServiceFabric.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `-1` :heavy_check_mark: |
| [Datadog.Trace.AppSec.Transports.Http.IpInfo](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/AppSec/Transports/Http/IpInfo.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.ClrProfiler.Helpers.PointerHelpers](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/ClrProfiler/Helpers/PointerHelpers.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.AppSec.InstrumentationGatewayEventArgs](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/AppSec/InstrumentationGatewayEventArgs.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpRequest](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/HttpOverStreams/HttpRequest.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpResponse](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/HttpOverStreams/HttpResponse.cs) | `-50%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.Transports.HttpStreamResponse](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/Agent/Transports/HttpStreamResponse.cs) | `-33%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.SpanContext](https://github.com/DataDog/dd-trace-dotnet/tree/01750fd782da38bb0e7f85779a153cbb8a0ff65f/tracer/src/Datadog.Trace/SpanContext.cs) | `-27%` :no_entry: | `0%` :heavy_check_mark:   | `2` :no_entry: |
| ...And 49 more  | | | |


The following classes were added in #2048:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.AppSec.AspNetCore.BlockingMiddleware](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `41%` | `19%` | `25` |
| [Datadog.Trace.AppSec.BlockActionException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.AppSec.PageBlockedByAppSecException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.Ci.CITracerManager](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `1` |
| [Datadog.Trace.Ci.CITracerManagerFactory](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `3` |
| ...And 36 more  | | | |


76 classes were removed from Datadog.Trace in #2048


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81337&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81337/artifacts?artifactName=Code%20Coverage%20Report_81337&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81545&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81545/artifacts?artifactName=Code%20Coverage%20Report_81545&api-version=6.0&%24format=zip)

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/4153faab84949d48bcec08c96e9d237dad63f508) will will **increase** line coverage by `4%`
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/4153faab84949d48bcec08c96e9d237dad63f508) will **not change** branch coverage
:heavy_check_mark: Merging #2048 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/4153faab84949d48bcec08c96e9d237dad63f508) will will **decrease** complexity by `589`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/4153faab84949d48bcec08c96e9d237dad63f508) | #2048       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `10490` / `15636` | `9092` / `12795` |          |
| Lines %   | `67%`      | `71%`      |  `4%` :heavy_check_mark:  |
| Branches  | `4585` / `6890` | `4055` / `6110` |          |
| Branches %| `67%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `8098`      | `7509`        |  `-589`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81540&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81540/artifacts?artifactName=Code%20Coverage%20Report_81540&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81566&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81566/artifacts?artifactName=Code%20Coverage%20Report_81566&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/4153faab84949d48bcec08c96e9d237dad63f508) | #2048       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `67%`      | `71%`       |  `4%` :heavy_check_mark:  |
| Branches %| `67%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `8098`      | `7509`       |  `-589` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.DuckTyping.DuckTypeProxyMethodsWithGenericParametersNotSupportedInNonPublicInstancesException](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.PlatformHelpers.ServiceFabric](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/PlatformHelpers/ServiceFabric.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `-1` :heavy_check_mark: |
| [Datadog.Trace.AppSec.Transports.Http.IpInfo](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/AppSec/Transports/Http/IpInfo.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.ClrProfiler.Helpers.PointerHelpers](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/ClrProfiler/Helpers/PointerHelpers.cs) | `-75%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.AppSec.InstrumentationGatewayEventArgs](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/AppSec/InstrumentationGatewayEventArgs.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpRequest](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/HttpOverStreams/HttpRequest.cs) | `-60%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.HttpOverStreams.HttpResponse](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/HttpOverStreams/HttpResponse.cs) | `-50%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.Transports.HttpStreamResponse](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/Agent/Transports/HttpStreamResponse.cs) | `-33%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.SpanContext](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/SpanContext.cs) | `-27%` :no_entry: | `0%` :heavy_check_mark:   | `2` :no_entry: |
| [Datadog.Trace.DuckTyping.DuckType](https://github.com/DataDog/dd-trace-dotnet/tree/a6a287ae249a2b7137c6d561778db7053e6961cc/tracer/src/Datadog.Trace/DuckTyping/DuckType.Properties.cs) | `-24%` :no_entry: | `-30%` :no_entry:   | `0` :heavy_check_mark: |
| ...And 49 more  | | | |


The following classes were added in #2048:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.AppSec.AspNetCore.BlockingMiddleware](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `41%` | `19%` | `25` |
| [Datadog.Trace.AppSec.BlockActionException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.AppSec.PageBlockedByAppSecException](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `0%` | `100%` | `4` |
| [Datadog.Trace.Ci.CITracerManager](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `1` |
| [Datadog.Trace.Ci.CITracerManagerFactory](https://github.com/DataDog/dd-trace-dotnet/pull/2048/files) | `100%` | `100%` | `3` |
| ...And 36 more  | | | |


77 classes were removed from Datadog.Trace in #2048


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81540&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81540/artifacts?artifactName=Code%20Coverage%20Report_81540&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2048](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=81566&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/81566/artifacts?artifactName=Code%20Coverage%20Report_81566&api-version=6.0&%24format=zip)

**Pull request merged**